test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -223,6 +226,161 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
+				}
+			}
+			logs.Infof("Released PodIP: %v, Interface: %s", podIP, iface)
+		}
+		return true, nil
+	})
+}
+
+func checkOVSPorts(before, after []string) bool {
+	afterSet := make(map[string]struct{})
+	for _, port := range after {
+		afterSet[port] = struct{}{}
+	}
+
+	for _, port := range before {
+		if _, exists := afterSet[port]; !exists {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (data *testData) getOVSPortsOnSecondaryBridge(t *testing.T, nodeName string) ([]string, error) {
+	cmd := []string{"ovs-vsctl", "list-ports", "br-secondary"}
+
+	stdout, stderr, err := data.e2eTestData.RunCommandFromAntreaPodOnNode(nodeName, cmd)
+	if err != nil {
+		return nil, fmt.Errorf("failed to run ovs-vsctl on node %s: %v\nstderr: %s", nodeName, err, stderr)
+	}
+
+	ovsPorts := strings.Fields(stdout)
+	return ovsPorts, nil
+}
+
+// reconcilationAfterAgentRestart verifies OVS Ports cleanup and IP release.
+func (data *testData) reconcilationAfterAgentRestart(t *testing.T) error {
+	nodes := []string{data.pods[0].nodeName, data.pods[2].nodeName}
+	for _, node := range nodes {
+		ovsPortsBeforeAgentRestart, err := data.getOVSPortsOnSecondaryBridge(t, node)
+		if err != nil {
+			return fmt.Errorf("failed to get Secondary bridge OVS Ports before agent restart: %w", err)
+		}
+
+		// Restarting the Antrea agent
+		if err := data.e2eTestData.RestartAntreaAgentPods(30 * time.Second); err != nil {
+			t.Fatalf("Failed to restart Antrea agent Pods: %v", err)
+		}
+
+		ovsPortsAfterAgentRestart, err := data.getOVSPortsOnSecondaryBridge(t, node)
+		if err != nil {
+			return fmt.Errorf("failed to get Secondary bridge OVS Ports after agent restart: %w", err)
+		}
+
+		if !checkOVSPorts(ovsPortsBeforeAgentRestart, ovsPortsAfterAgentRestart) {
+			return fmt.Errorf("Secondary bridge OVS Ports mismatch after agent restart: %w", err)
+		}
+	}
+	return data.deleteVLANPod(t)
+}
+
+func (data *testData) deleteVLANPod(t *testing.T) error {
+	vlanPod := data.pods[1]
+	ifaces := []string{"eth1", "eth2"}
+	ifacesIPs, err := data.listPodIPs(vlanPod)
+	if err != nil {
+		return fmt.Errorf("failed to get IPs of Interfaces: %w", err)
+	}
+
+	ovsPortsBeforeDeletion, err := data.getOVSPortsOnSecondaryBridge(t, vlanPod.nodeName)
+	if err != nil {
+		return fmt.Errorf("failed to get OVS Ports Before Pod deletion: %w", err)
+	}
+
+	ipPools, err := data.getIPPoolNames(ifaces, vlanPod, "default")
+	if err != nil {
+		return fmt.Errorf("failed to get IPPool: %w", err)
+	}
+
+	// Remove Pod and check IP released or not.
+	if err := data.e2eTestData.DeletePodAndWait(defaultTimeout, vlanPod.podName, data.e2eTestData.GetTestNamespace()); err != nil {
+		return fmt.Errorf("failed to delete Pod %s: %w", vlanPod.podName, err)
+	}
+
+	ovsPortsAfterDeletion, err := data.getOVSPortsOnSecondaryBridge(t, vlanPod.nodeName)
+	if err != nil {
+		return fmt.Errorf("failed to get OVS Ports After Pod deletion: %w", err)
+	}
+
+	if checkOVSPorts(ovsPortsBeforeDeletion, ovsPortsAfterDeletion) {
```

Comment:
Could we check the exact the OVS port was deleted? We can look up the port by the name right?

Reply:
Ok, but I think the names are generated with a deterministic method (by hashing Pod + interface name, I think). So, you can run the test once and find out the port names got deleted e.g. from a debug log, and then use the names in your code?

Reply:
we discussed it offline OVS port name generated way is not deterministic , it takes containerID as an input  so the generated name can change.